server.js

```diff
@@ -16,9 +16,155 @@ const app = express();
+    type: String,
+    required: [true, "Please type a message."],
+    minLength: [
+      5,
+      "Your text must be minimum 5 characters and it was '{VALUE}' characters.",
+    ],
+    maxLength: [
+      140,
+      "Your text must be max 140 characters and it was '{VALUE}' characters.",
+    ],
+    trim: true,
+  },
+  likes: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date(),
+  },
+  ex: {
+    type: String,
+    required: [true, "Please type your ex's name."],
+    minLength: [
+      4,
+      "Your ex's name must be minimum 4 characters and it was '{VALUE}' characters.",
+    ],
+    maxLength: [
+      15,
+      "Your ex's name must be max 15 characters and it was '{VALUE}' characters.",
+    ],
+    trim: true,
+  },
+});
+
+const Text = mongoose.model("Message", TextSchema);
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send([
+    { "/": "Hi! This is the text your ex backend" },
+    { "https://textyourex.netlify.app/": "Frontend here!" },
+  ]);
+});
+
+// GET ALL TEXTS - LIMIT AT 20
+app.get("/texts", async (req, res) => {
+  try {
+    const texts = await Text.find()
+      .sort({ createdAt: "desc" })
+      .limit(40)
+      .exec();
+    res.status(200).json({ success: true, data: texts });
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      status_code: 400,
+      message: "Bad request, could not fetch thoughts.",
+      error: err.errors,
+    });
+  }
+});
+
+// POST A NEW TEXT
+app.post("/texts", async (req, res) => {
+  const { message, ex } = req.body;
+  try {
+    const newText = await new Text({
+      message: message,
+      ex: ex.toLowerCase(),
+    }).save();
+    res.status(201).json(newText);
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      status_code: 400,
+      message: "Bad request, could not save this new thought.",
+      error: err.errors,
+    });
+  }
+});
+
+// POST A LIKE TO THE TEXT
+app.post("/texts/:textId/like", async (req, res) => {
+  const { textId } = req.params;
+  try {
+    const likeText = await Text.findByIdAndUpdate(textId, {
+      $inc: { likes: 1 },
+    });
+    res.status(201).json(likeText);
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      status_code: 400,
+      message: "Bad request, could not find and update this thought.",
+      error: err.errors,
+    });
+  }
+});
+
+// RETURN MOST LIKED TEXTS - 20 TEXTS
+app.get("/mostliked", async (req, res) => {
+  try {
+    const texts = await Text.find().sort({ likes: "desc" }).limit(20).exec();
+    res.status(200).json({ success: true, data: texts });
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      status_code: 400,
+      message: "Bad request, could not fetch thoughts.",
+      error: err.errors,
+    });
+  }
+});
+
+// SEARCH FOR EX TEXTS BY NAME - LIMIT TO 20 TEXTS - THIS ISN'T WORKING
```

Comment:
Too bad it isn't working, it would be a nice feature. But you'll definitely be able to implement it later on if you like!